waspc/cli/src/Wasp/Cli/Command/Build.hs

```diff
@@ -58,40 +73,65 @@ build = do
+    prepareFilesNecessaryForDockerBuild waspProjectDir buildDir = runExceptT $ do
+      -- Until we implement the solution described in https://github.com/wasp-lang/wasp/issues/1769,
+      -- we're copying all files and folders necessary for the build into the .wasp/build directory.
+      -- We chose this approach for 0.12.0 (instead of building from the project root) because:
+      --   - The build context remains small (~1.5 MB vs ~900 MB).
+      --   - We don't risk copying possible secrets from the project root into the build context.
+      --   - The commands for building the project stay the same as before
+      --     0.12.0, which is good for both us (e.g., for fly deployment) and our
+      --     users  (no changes in CI/CD scripts).
+      -- For more details, read the issue linked above.
+      liftIO $
+        copyDirectory
+          (waspProjectDir </> srcDirInWaspProjectDir)
+          (buildDir </> castRel srcDirInWaspProjectDir)
+
+      liftIO $
+        copyDirectory
+          (waspProjectDir </> dotWaspDirInWaspProjectDir </> generatedCodeDirInDotWaspDir </> sdkRootDirInGeneratedCodeDir)
+          (buildDir </> sdkRootDirInGeneratedCodeDir)
+
+      let packageJsonInBuildDir = buildDir </> castRel packageJsonInWaspProjectDir
+      let packageLockJsonInBuildDir = buildDir </> castRel packageLockJsonInWaspProjectDir
+
+      liftIO $
+        copyFile
+          (waspProjectDir </> packageJsonInWaspProjectDir)
+          packageJsonInBuildDir
+
+      liftIO $
+        copyFile
+          (waspProjectDir </> packageLockJsonInWaspProjectDir)
+          packageLockJsonInBuildDir
 
-      copyDirectory
-        (waspProjectDir </> dotWaspDirInWaspProjectDir </> generatedCodeDirInDotWaspDir </> sdkRootDirInGeneratedCodeDir)
-        (buildDir </> sdkRootDirInGeneratedCodeDir)
+      -- A hacky quick fix for https://github.com/wasp-lang/wasp/issues/2368
+      -- We should remove this code once we implement a proper solution.
+      ExceptT $ updateJsonFile removeWaspConfigDevDependency packageJsonInBuildDir
+      ExceptT $ updateJsonFile removeAllMentionsOfWaspConfig packageLockJsonInBuildDir
 
-      copyFile
-        (waspProjectDir </> packageJsonInWaspProjectDir)
-        (buildDir </> castRel packageJsonInWaspProjectDir)
+    removeAllMentionsOfWaspConfig :: Value -> Value
+    removeAllMentionsOfWaspConfig packageLockJsonObject =
+      packageLockJsonObject
+        & key "packages" . key "" %~ removeWaspConfigDevDependency
+        & key "packages" . _Object
+          %~ HM.filterWithKey
+            (\packageKey _ -> not ("wasp-config" `isInfixOf` packageKey))
 
-      copyFile
-        (waspProjectDir </> packageLockJsonInWaspProjectDir)
-        (buildDir </> castRel packageLockJsonInWaspProjectDir)
+    removeWaspConfigDevDependency :: Value -> Value
+    removeWaspConfigDevDependency original =
+      original & key "devDependencies" . _Object . at "wasp-config" .~ Nothing
```

Comment:
This is where all the lens stuff is.

Reply:
Ok, this is readable I think! I do wonder if there is not sometihng, at lens level, that would allow you to unset `wasp-config` field, instead of setting it to Nothing? I guess this works, so not a biggie, but that would be even more readable + a bit more direct. If not, no worries. I mean it is a key in hashmap so I am sure it is doable hm.

Reply:
I couldn't find it, and I don't think it's worth exploring further since this is a temporary fix and we want to get it out ASAP.

If that's ok with you, please resolve

Reply:
Yup, good with me.